bazaarvoice/emodb
# Added ~table, ~placement, and table attributes as S3 object tags for Stash files

## Github Issue #

None

## What Are We Doing Here?

This update uses S3 object tagging to tag the table and placement intrinsics as well as any table attributes as tags on Stash objects.  Functionally this makes no difference but it allows the creation of AWS policies which have permissions similar to Emo's own read permissions.  Long term Emo itself may be able to regulate AWS roles with permissions that mimic those of its API keys, but for now this lays the groundwork for that possibility and allows for external actors to create their own AWS policies from these tags.

## How to Test and Verify

1. Check out this PR
2. Perform a Stash run to S3
3. Verify each Stash file is tagged with the table, placement and any table metadata associated with the table.

## Risk

Since the content of each Stash file is unchanged this is low risk.  The only real risk is if a defect was introduced that prevents Stash from being generated, such as an unexpected error response from S3 due to a problem with the tags.

### Level 

`Low`

### Required Testing

Mostly `Regression`, but `Manual` verification is required to ensure the S3 object tags are correctly set.

## Code Review Checklist


build and run.

we inevitably come back to this code it will only take hours to figure out, not
days.

a victim of rampaging consistency, and should be using this course of action. 
We don't have coding standards out yet for this project, so please make sure to address any feedback regarding STYLE so the codebase remains consistent.
